pkg/assembler/backends/inmem/certifyVuln.go

```diff
@@ -212,6 +213,21 @@ func (c *demoClient) CertifyVuln(ctx context.Context, filter *model.CertifyVulnS
 			foundOne = true
 		}
 	}
+	if !foundOne && filter != nil && filter.Vulnerability != nil &&
+		filter.Vulnerability.NoVuln != nil && *filter.Vulnerability.NoVuln {
+
+		exactVuln, err := c.exactVulnerability(&model.VulnerabilitySpec{
+			Type:            ptrfrom.String(noVulnType),
+			VulnerabilityID: ptrfrom.String(""),
+		})
+		if err != nil {
+			return nil, gqlerror.Errorf("%v :: %v", funcName, err)
+		}
+		if exactVuln != nil {
+			search = append(search, exactVuln.certifyVulnLinks...)
+			foundOne = true
+		}
+	}
```

Comment:
this should be else if - we don't need to go into the next branch if it is novuln